dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/LoopUnrolling.java

```diff
@@ -119,21 +120,22 @@ private void unrollLoop(CondJump loopBackJump, int bound) {
         int iterCounter = 0;
         while (++iterCounter <= bound) {
             if (iterCounter == bound) {
-                // Mark end of loop
-                loopBackJump.insertAfter(
-                        EventFactory.newStringAnnotation(String.format("// End of Loop: %s", loopBegin.getName())));
-
                 // Update loop iteration label
-                loopBegin.setName(String.format("%s%s%d", loopBegin.getName(), LOOP_ITERATION_SEPARATOR, iterCounter));
+                final String loopName = loopBegin.getName();
+                loopBegin.setName(String.format("%s%s%d", loopName, LOOP_ITERATION_SEPARATOR, iterCounter));
                 loopBegin.addFilters(Tag.NOOPT);
 
                 // This is the last iteration, so we replace the back jump by a bound event.
                 final Label threadExit = (Label) loopBackJump.getThread().getExit();
                 final CondJump boundEvent = EventFactory.newGoto(threadExit);
-                boundEvent.addFilters(loopBackJump.getFilters()); // Keep tags of original jump.
```

Comment:
Why were we doing this before and it is not needed now?

Reply:
We were doing this before because `LoopUnrolling` was responsible for keeping the `SPINLOOP`-tag that was generated by `FindSpinLoops` (now `SimpleSpinLoopDetection`). This gave the code more responsibility than it needed(*).
However, now `DynamicPureLoopCutting` is the only pass that places spinning related jumps.

(*) In fact, `LoopUnrolling` and `SimpleSpinLoopDetection` are still somewhat coupled by the fact that `LoopUnrolling` looks for the `SPINLOOP` tag to understand how many times to unroll the loop.
I would like to get rid of this coupling as well by letting `SimpleSpinLoopDetection` produce a `LoopBoundAnnotation(1)` in front of the spin loop, so that `LoopUnrolling` does not need to be aware of `Tag.SPINLOOP` at all and just unrolls everything "as usual".